### ISSUE TITLE
form-control: missing cursor if switching off readonly on dblclick [Chrome+Safari]

### ISSUE THREAD

**Issue by DenisVuyka** [twbs/bootstrap#16022]
The issue can be observed here: [jsfiddle](https://jsfiddle.net/dvuyka/htua4ugm/)
Was reproduced with Chrome 41.

Description: when switching off 'readonly' state of the **input.form-control** on the fly (via **dblclick** event) element stops showing text caret, however user can still type text. The issue is not reproduced for standard input controls without **form-control** class (see jsfidlle link above).

**Comment by twbs-lmvtfy** [comment]
Hi @DenisVuyka!

You appear to have posted a live example (https://fiddle.jshell.net/dvuyka/htua4ugm/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 52, column 73: Bad value ```` true ```` for attribute ```` readonly ```` on element ```` input ````.
* line 57, column 52: Bad value ```` true ```` for attribute ```` readonly ```` on element ```` input ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by DenisVuyka** [comment]
The example was updated with **readonly="readonly"**, this does affect the issue though.

**Comment by cvrebert** [comment]
Reposting so as to trigger the validator again: https://jsfiddle.net/dvuyka/htua4ugm/

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (https://fiddle.jshell.net/dvuyka/htua4ugm/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 12, column 3: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.2.0
* line 13, column 3: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.2.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
Corrected to use latest Bootstrap: https://jsfiddle.net/w16tvcp2/1/

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (https://fiddle.jshell.net/w16tvcp2/1/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 12, column 3: [E007](https://github.com/twbs/bootlint/wiki/E007): Only one copy of Bootstrap's JS should be included; currently the webpage includes both bootstrap.js and bootstrap.min.js
* line 22, column 7: [E007](https://github.com/twbs/bootlint/wiki/E007): Only one copy of Bootstrap's JS should be included; currently the webpage includes both bootstrap.js and bootstrap.min.js
* line 12, column 3: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.2.0
* line 13, column 3: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.2.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
Corrected to use latest Bootstrap: https://jsfiddle.net/w16tvcp2/2/

**Comment by cvrebert** [comment]
OS X Safari is also affected.

**Comment by cvrebert** [comment]
Simplified testcase: https://jsfiddle.net/f8utvj06/2/
Apparently this somehow relates to `outline`...

**Comment by DenisVuyka** [comment]
@cvrebert Thanks for simplified scenario. I've noticed that control starts working when setting "outline: -2px;". I guess this is something to be fixed at 'form-control' side.

**Comment by cvrebert** [comment]
Filed Chrome bug: https://code.google.com/p/chromium/issues/detail?id=465274

**Comment by cvrebert** [comment]
Discovered a different Firefox bug in the course of investigating your bug; reported it as https://bugzilla.mozilla.org/show_bug.cgi?id=1141011

**Comment by cvrebert** [comment]
Filed WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=142477
Filed Safari Radar: http://openradar.appspot.com/20090003

**Comment by cvrebert** [comment]
Note that this isn't `outline:0`-specific. Chrome has the same problem with `#input1:focus { outline: 5px solid red; }`.
@mdo Any thoughts?

**Comment by zacechola** [comment]
These need to be added to the wall of bugs, @cvrebert?

**Comment by cvrebert** [comment]
@zacechola The Chrome bug, yes. The others aren't Bootstrap-specific.

**Comment by mdo** [comment]
Anything left to do here?

**Comment by cvrebert** [comment]
@mdo Unless you happen to know some relevant CSS magic or were to think this bug is more important than the custom `:focus` `outline`, I think this is CantFix/WontFix due to underlying browser bugs.

**Comment by amir143979** [comment]
hfdhfhfds
